src/json.ts

```diff
@@ -0,0 +1,355 @@
+/**
+ * This file contains a set of utility functions for working with TypedMap<string, JSONValue>
+ * objects in AssemblyScript. These utilities provide a way to create, update, and convert
+ * TypedMaps to and from JSON strings. Additionally, they offer functionality for adding,
+ * removing, or updating TypedMap entries and performing operations on arrays within TypedMaps.
+ *
+ * Due to AssemblyScript's limitations when working with JSON, TypedMap<string, JSONValue> is
+ * used as a close approximation of a regular JavaScript object. These utility functions aim
+ * to bridge the gap, making it easier to work with TypedMaps in a similar way to how one
+ * would work with objects in JavaScript.
+ */
```

Comment:
Should we have a test file that specifically just has tests for this utility library itself? @yoshiwarab

Reply:
yeye, we do, see json.test.ts!

Reply:
AH – missed this, thank you 🤦 🙏